### PR TITLE
Add favourite API for osu! client

### PR DESCRIPTION
**Pull request by cmyui**
Proposed SQL table: 
```sql
CREATE TABLE user_favorites (
    user_id INTEGER NOT NULL,
    beatmapset_id INTEGER NOT NULL,
    created_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
    PRIMARY KEY (user_id, beatmapset_id)
);
```

### CODE REVIEW

File: app/api/favorites.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+from fastapi import Depends
+from fastapi import Query
+
+import app.usecases
+from app.models.user import User
+from app.usecases.user import authenticate_user
+
+
+async def add_favorite(
+    user: User = Depends(authenticate_user(Query, "u", "h")),
```
Comment (cmyui) [inline app/api/favorites.py]: Confirmed params via fiddler

Comment (cmyui) [reply app/api/favorites.py]: Sorry for no image i closed it

Comment (cmyui) [reply app/api/favorites.py]: https://github.com/osuAkatsuki/bancho.py/blob/e8817c8addf733e1e0b873f7b66aabd41cb98254/app/api/domains/osu.py#L247-L262

File: app/usecases/favorites.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import app.state.services
+
+
+async def exists(user_id: int, beatmapset_id: int) -> bool:
+    rec = await app.state.services.database.fetch_one(
+        """\
+        SELECT 1
+        FROM user_favourites
+        WHERE user_id = :user_id
+        AND setid = :beatmapset_id
```
Comment (cmyui) [inline app/usecases/favorites.py]:
```suggestion
        AND beatmapset_id = :beatmapset_id
```

File: app/api/favorites.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+) -> bytes:
+    if await app.usecases.favorites.exists(user.id, map_set_id):
+        return b"You've already favourited this beatmap!"
+
+    await app.usecases.favorites.add_favorite(
+        userid=user.id,
+        setid=map_set_id,
+    )
+
+    return b"Added favourite!"
```
Comment (infernalfire72) [inline app/api/favorites.py]: pick one :trolley:

File: app/usecases/favorites.py (hunk 2 above)
Comment (infernalfire72) [inline app/usecases/favorites.py]: buggus

**Review by infernalfire72 — APPROVED** [review]

**Pull request merged**